plugins/vpc-eni/config/netconfig.go

```diff
@@ -43,6 +44,7 @@ type netConfigJSON struct {
 	ENIIPAddresses     []string `json:"eniIPAddresses"`
 	GatewayIPAddresses []string `json:"gatewayIPAddresses"`
 	UseExistingNetwork bool     `json:"useExistingNetwork"`
+	BlockIMDS          bool     `json:"block-instance-metadata"`
```

Comment:
Let's name consistently. camelCase instead of '-'s.

I am assuming this is configurable because on EC2 we will allow IMDS access while we disallow it on Fargate?

Reply:
Yes this can be configured for EC2. Based on an environment variable `ECS_AWSVPC_BLOCK_IMDS` set by the user, this particular config would be set for the CNI plugin by the agent.